File: leetcode/questions/27/python/27.py
class Solution(object):
    def removeElement(self, nums, val):
        last = len(nums) - 1

        for idx in reversed(range(len(nums))):
            if nums[idx] == val:
                nums[idx], nums[last] = nums[last], nums[idx]
                last -= 1
        
        return last + 1

File: leetcode/questions/27/java/27.java
class Solution {
    private void swap(int[] nums, int idxA, int idxB) {
        int aux = nums[idxA];
        nums[idxA] = nums[idxB];
        nums[idxB] = aux;
    }

    public int removeElement(int[] nums, int val) {
        int last = nums.length - 1;

        for (int idx = nums.length - 1; idx >= 0; --idx) {
            if (nums[idx] == val) {
                swap(nums, idx, last);
                last -= 1;
            }
        }

        return last + 1;
    }
}